src/compiler/checker.ts

```diff
@@ -1107,6 +1107,35 @@ namespace ts {
             return diagnostic;
         }
 
+        function addDeprecatedSuggestionImpl(declarations: Node[], diagnostic: DiagnosticWithLocation) {
+            for (const declaration of declarations) {
+                const deprecatedTag = getJSDocDeprecatedTag(declaration);

```

Comment:
re @gurunars suggestion:

```suggestion
            const deprecatedTag = forEach(declarations, getJSDocDeprecatedTag);
```

I think `Array.find` and our similar `find` function would also work. though the types are less permissive than `forEach`.